src/cmd-virt-install

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+# "install" a coreos system with libvirt.  This code in the future
+# ideally gets added to `virt-install`.  See also https://github.com/coreos/fedora-coreos-tracker/issues/235
+# It is inspired by https://coreos.com/os/docs/latest/booting-with-libvirt.html
+
+import argparse
+import subprocess
+import json
+import os
+import re
+import sys
+import random
+import string
+import tempfile
+
+import libvirt
+
+cosa_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, cosa_dir)
+
+from cosalib import cmdlib
+from cosalib.builds import Builds
+
+# Parse args and dispatch
+parser = argparse.ArgumentParser()
+parser.add_argument("--connect", "-c", help="libvirt URI", default="qemu:///system")
+parser.add_argument("--ignition", "-i", help="Ignition config", required=True)
+parser.add_argument("--build", help="Build ID")
+parser.add_argument("--pool", help="storage pool", default="default")
+parser.add_argument("--name", help="Prefix for libvirt resources")
+parser.add_argument("--instid", help="Suffix appended to libvirt resources")
+parser.add_argument("vinstall_args", nargs='*', help="uffix appended to libvirt resources")
+args = parser.parse_args()
+
+builds = Builds()
+if args.build is None:
+    args.build = builds.get_latest()
+builddir = builds.get_build_dir(args.build)
+with open(os.path.join(builddir, "meta.json")) as f:
+    buildmeta = json.load(f)
+qemuimg = buildmeta.get('images', {}).get('qemu')
+if qemuimg is None:
+    raise SystemExit(f"No qemu image in build: {args.build}")
+qemupath = os.path.join(builddir, qemuimg['path'])
+ignition_version = cmdlib.disk_ignition_version(qemupath)
+
+if args.name is None:
+    args.name = buildmeta['name']
+
+if args.instid is None:
+    args.instid = ''.join([random.choice(string.ascii_letters + string.digits) for n in range(8)])
+
+vconn = libvirt.open(args.connect)
+if vconn is None:
+    raise Exception('Failed to connect to libvirt "{}"'.format(args.connect))
+
+basevolname = os.path.basename(qemupath)
+
+
+def virsh(*vargs):
+    subprocess.check_call(['virsh', '-c', args.connect] + list(vargs))
+
+
+pool = vconn.storagePoolLookupByName(args.pool)
+
+
+def get_pool_path(connect, pool):
+    poolxml = subprocess.check_output(['virsh', '-c', connect, 'pool-dumpxml', pool], encoding='UTF-8')
+    r = re.compile(re.escape("<path>") + "(.*)" + re.escape("</path>"))
+    for line in poolxml.split():
+        m = r.search(line)
+        if not m:
+            continue
+        return m.group(1)
+    raise Exception(f"Failed to find path for pool: {pool}")
+
+
+poolpath = get_pool_path(args.connect, args.pool)
+
+qemu_size = str(json.loads(subprocess.check_output(['qemu-img', 'info', '--output=json', qemupath]))['virtual-size'])
```

Comment:
Could the size be read from `meta.json` for the qemu artifact instead?

Reply:
This is the virtual size, distinct from the qemu file's size.